src/Sylius/Bundle/ThemeBundle/Asset/Installer/OutputAwareAssetsInstaller.php

```diff
@@ -0,0 +1,99 @@
+     */
+    public function setOutput(OutputInterface $output)
+    {
+        $this->output = $output;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function installAssets($targetDir, $symlinkMask)
+    {
+        if ($this->hasOutput()) {
+            if (self::HARD_COPY !== $symlinkMask) {
+                $this->output->writeln('Trying to install assets as <comment>symbolic links</comment>.');
+            } else {
+                $this->output->writeln('Installing assets as <comment>hard copies</comment>.');
+            }
+        }
+
+        return parent::installAssets($targetDir, $symlinkMask);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function installBundleAssets(BundleInterface $bundle, $targetDir, $symlinkMask)
+    {
+        $sources = $this->findAssetsPathsForBundle($bundle);
+
+        if (0 === count($sources)) {
+            $effectiveSymlinkMask = $symlinkMask;
+        } else {
+            if ($this->hasOutput()) {
+                $this->output->writeln(sprintf('Installing assets for <comment>%s</comment>', $bundle->getNamespace(), $targetDir));
+            }
+
+            $effectiveSymlinkMask = parent::installBundleAssets($bundle, $targetDir, $symlinkMask);
+
+            if ($this->hasOutput()) {
+                if (self::RELATIVE_SYMLINK === $symlinkMask) {
+                    if (self::SYMLINK === $effectiveSymlinkMask) {
+                        // Wanted to install assets using relative symbolic links, but used absolute symbolic links.
+                        $this->output->writeln('It looks like your system doesn\'t support relative symbolic links, so the assets were installed by using absolute symbolic links.');
+                        return $effectiveSymlinkMask;
+                    }
+                }
+
+                if (self::HARD_COPY !== $symlinkMask) {
+                    if (self::HARD_COPY !== $effectiveSymlinkMask) {
+                        // Wanted to install assets using symbolic links (relative or absolute), and it was successful.
+                        $this->output->writeln('The assets were installed using symbolic links.');
+                        return $effectiveSymlinkMask;
+                    } else {
+                        // Wanted to install assets using symbolic links, but they was copied.
+                        $this->output->writeln('It looks like your system doesn\'t support symbolic links, so the assets were installed by copying them.');
+                        return $effectiveSymlinkMask;
+                    }
+                }
+
+                if (self::HARD_COPY === $symlinkMask) {
+                    if (self::HARD_COPY === $effectiveSymlinkMask) {
+                        // Wanted to install assets by copying, and it was successfull.
+                        $this->output->writeln('The assets were copied.');
+                        return $effectiveSymlinkMask;
+                    }
+                }
```

Comment:
Extract to method

```php
$messageMask = $symlinkMask + $effectiveSymlinkMask;
switch($messageMask) {
    case 0:
               $message = 'The assets were copied.';
               break;
    ....
    case 2:
               if ($symlinkMask === $effectiveSymlinkMask) {
                    $message = 'Used symlinks';
               } else {
                    $message = 'Tried relative symlinks, but used hard copy';
               }
               break;
     ....
}
$this->output->writeln($message);
```

```$messageMask``` can have the following values:

| messageMask             | symlinkMask | effectiveSymlinkMask
| ------------- | ---| -----
|0|0/H|0/H
|1|1/S|0/H
|2|1/S|1/S
|2|2/R|0/H
|3|2/R|1/S
|4|2/R|2/R

So we can easily decide which message to write.